q1_06.ts

```diff
@@ -0,0 +1,21 @@
+// 1.06 (*) Find out whether a list is a palindrome.
+import { reverseList } from "./q1_05";
+const notPalindromeList = [1, 2, 3, 4, 5, 21, 75];
+const palindromeList = [1, 2, 3, 2, 1];
+
+export function checkPalindrome<T>(input: Array<T>) {
+  // if (reverseList(input) === input) {
+  //   return true;
+  // } else {
+  //   return false;
+  // }
+
+  if (JSON.stringify(reverseList(input)) === JSON.stringify(input)) {
```

Comment:
Whats the purpose of JSON.stringify here ? Can you come up with a better way to reduce time complexity of this function? Right now you are checking equality for every character, therefore its time complexity is appx O(n) where n is string length that too ignoring reverse and JSON.stringify operation.